### PR TITLE
[AppBar] Add a Typography themer.

### PR DESCRIPTION
**Pull request by jverkoey**
Pivotal story: https://www.pivotaltracker.com/story/show/156769436

Before: 
before

After:
simulator screen shot - iphone 8 plus - 2018-04-13 at 15 05 25

### CODE REVIEW

File: components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (ianegordon) [inline components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h]: s/2017/2018/
here an below.

File: components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#import "MaterialAppBar.h"
+#import "MaterialTypographyScheme.h"
+
+/**
+ A Typography themer for MDCAppBar that implements the Material design Typography system mappings.
+ */
+@interface MDCAppBarTypographyThemer : NSObject
+
+/**
+ Applies a Typography scheme's properties to an MDCAppBar.
```
Comment (ianegordon) [inline components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h:26]: Consider noting that this themer essentially wires into the underlying NavigationBar.

**Review by ianegordon — DISMISSED** [review]
Aside from date C&P issue, LGTM.
Feel free to merge after fix.

File: components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h (hunk 1 above)
Comment (jverkoey) [reply components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h]: Done.

File: components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h (hunk 2 above)
Comment (jverkoey) [reply components/AppBar/src/TypographyThemer/MDCAppBarTypographyThemer.h:26]: I think that's an implementation detail that we can probably leave out of the comments.

**Comment by jverkoey** [comment]
This is blocked on a fix to navigation bar.

**Comment by jverkoey** [comment]
Blocked by https://github.com/material-components/material-components-ios/pull/3360

**Review by ianegordon — APPROVED** [review]
LGTM.
Please merge when blocker is cleared.

**Comment by jverkoey** [comment]
Rebased off develop with the fixes introduced in #3360.

**Pull request merged**